src/v/datalake/translation/deps.cc

```diff
@@ -437,19 +436,27 @@ class partition_translation_context : public translation_context {
     void reconcile_properties() final {
         auto old_invalid_action = _invalid_record_action;
         auto old_cp_enabled = _cp_enabled;
-        auto old_target_lag = _target_lag;
 
+        // target lag changes will be picked dynamically and does not
+        // need a state reset.
         _invalid_record_action = compute_invalid_record_action();
         _cp_enabled = translation_task::custom_partitioning_enabled{
           _features.is_active(features::feature::datalake_iceberg_ga)};
-        _target_lag = compute_target_lag();
 
-        if (_in_progress_translation) {
+        auto properties_updated = old_invalid_action != _invalid_record_action
+                                  || old_cp_enabled != _cp_enabled;
+        if (properties_updated) {
```

Comment:
Do we have a test validating this ?

Reply:
Yes, currently in datalake_upgrade_test that flips custom_partition_enabled after upgrade triggereing a rest of translation.